### PR TITLE
Minimum number of deletion to make it a palindrome in C++

### PR DESCRIPTION
**Pull request by Meghag5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes

### Description

(Write your answer here.)
This problem will be solved using dp.
We will be given one strings and we have to print the
Minimum number of deletion in a string to make it a palindrome.
Input - 
Enter string X
bbbab
Output -  
Minimum number of deletion in a string to make it a palindrome
4

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#5757

### CODE REVIEW

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+int main() {
+    string X;
+    cin>>X;
+    string Y = X;
+    reverse(Y.begin(),Y.end());
+    int m = X.length();
+    int n = Y.length();
+    //memset(t,-1,sizeof(t));
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: If it has no significance in the code, remove it up

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ This problem will be solved using dp.
+ We will be given one strings and we have to print the
+ Minimum number of deletion in a string to make it a palindrome.
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+int t[1001][1001];
+
+int longestSubsequence(string X,string Y,int m,int n){
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: Can you use more meaningful variable names?

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                t[i][j] = max(t[i-1][j], t[i][j-1]);
+            }
+        }
+    }
+    
+    return m - t[m][n];
+}
+
+int main() {
+    string X;
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: Can you use more meaningful variable names?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
@HarshCasper pr updated pls check

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format the entire Code

File: C-Plus-Plus/README.md [hunk 4]
```diff
@@ -433,6 +433,7 @@ _add list here_
 - [ZeroN Knapsack](dp/Zero_N_knapsack.cpp)
 - [K-Tree](dp/K-Tree.cpp)
 - [Sprague Grundy Theorem on Strings](dp/Sprague_Grundy_Theorem.cpp)
+- [Minimum number of deletion in a string to make it a palindrome](dp/Minimum_number_of_deletion _to_make_palindrome.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: The file path is buggy here. Kindly fix it

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+  }
+
+  for (int i = 1; i < m + 1; i++) {
+    for (int j = 1; j < n + 1; j++) {
+      //if letters of both string are equal then we will decrease row and column both by 1
+      if (X[i - 1] == Y[j - 1]) {
+        dp[i][j] = dp[i - 1][j - 1] + 1;
+      }
+      //if letters of both string are not equal then first we will decrease row by 1 and then column by 1 and find maximum value b/w two
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: Break down this comment. Too difficult to make any sense of

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  for (int i = 0; i < m + 1; i++) {
+    for (int j = 0; j < n + 1; j++) {
+      if (i == 0 || j == 0)
+        dp[i][j] = 0;
+    }
+  }
+
+  for (int i = 1; i < m + 1; i++) {
+    for (int j = 1; j < n + 1; j++) {
+      //if letters of both string are equal then we will decrease row and column both by 1
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: Break down this comment. Too difficult to make any sense of

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md [hunk 7]
```diff
@@ -433,6 +433,7 @@ _add list here_
 - [ZeroN Knapsack](dp/Zero_N_knapsack.cpp)
 - [K-Tree](dp/K-Tree.cpp)
 - [Sprague Grundy Theorem on Strings](dp/Sprague_Grundy_Theorem.cpp)
+- [Form a palindrome ](dp/Minimum_number_of_deletion _to_make_palindrome.cpp)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/README.md]: `[Minimum number of deletion to make palindrome]` rewrite it like this

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+ We will be given one strings and we have to print the
+ Minimum number of deletion in a string to make it a palindrome.
+*/
+
+#include <bits/stdc++.h>
+
+using namespace std;
+int dp[1001][1001];
+
+int minDeletion(string X, string Y, int m, int n) {
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: don't use single letters for variables use meaningful names

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+int main() {
+  string s1;
+  cin >> s1;
+  string s2 = s1;
+  reverse(s2.begin(), s2.end());
+  int m = s1.length();
+  int n = s2.length();
+  cout << minDeletion(s1, s2, m, n);
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: Use proper display message how you are reading and writing outputs

File: C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+/*
+Input - 
+Enter string X
+bbbab
+Output -  
+Minimum number of deletion in a string to make it a palindrome
+4
+Time Complexity - O(m*n)
+Space Complexity - O(m*n)
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_number_of_deletion _to_make_palindrome.cpp]: Use proper spacing

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
Done all the changes required

File: C-Plus-Plus/README.md [hunk 11]
```diff
@@ -433,6 +433,7 @@ _add list here_
 - [ZeroN Knapsack](dp/Zero_N_knapsack.cpp)
 - [K-Tree](dp/K-Tree.cpp)
 - [Sprague Grundy Theorem on Strings](dp/Sprague_Grundy_Theorem.cpp)
+- [Minimum number of deletion to make palindrome](dp/Minimum_number_of_deletion _to_make_palindrome.cpp)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/README.md]: You file is buggy due to inappropriate naming of file `(dp/Minimum_number_of_deletion _to_make_palindrome.cpp)`

Rename it properly or make it as `(dp/Minimum_number_of_deletion.cpp)`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
Done @ankitaggarwal23

File: C-Plus-Plus/dp/Minimum_number_of_deletion [hunk 12]
```diff
@@ -0,0 +1,61 @@
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Minimum_number_of_deletion]: Remove this line

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
You have to save your file in .cpp you didn't save it properly

**Comment by Meghag5** [comment]
Really sorry @ankitaggarwal23 
done

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**